pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -856,7 +856,7 @@ func TestGarbageCollectorSync(t *testing.T) {
 	stopCh := make(chan struct{})
 	defer close(stopCh)
 	go gc.Run(1, stopCh)
-	go gc.Sync(fakeDiscoveryClient, 10*time.Millisecond, stopCh)
+	go gc.Sync(fakeDiscoveryClient, 100*time.Millisecond, stopCh)
```

Comment:
Isn't this 100 ms timeout also going to be flaky, given that the first check also happens after 100ms?

Reply:
yes. we should not do anything in tests with such tight time tolerances. the 10ms sync loop period was in contrast to the 1s wait to see if progress was made (e.g. 100x difference between loop period and test.

I'm trying to understand what change is being made in this PR that broke this test... is that going to cause problems in real life as well?

Reply:
The old `WaitFor` will run the condition check function at least once more after the `done` channel is closed. The new behavior will most likely return a timeout error, though due to the nature of `select`, it has a slim chance to run the condition function once more.

I doubt any user would intentionally rely on the old behavior. Also, the old behavior wasn't explicitly documented. Hence, I think the chance to cause real problems is slim.